x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -193,6 +195,116 @@ export const getTupleDuplicateErrorsAndUniqueRules = (
   return [Array.from(errors.values()), Array.from(rulesAcc.values())];
 };
 
+// functions copied from here
+// https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/server/saved_objects/import/lib/check_origin_conflicts.ts#L55-L57
+const createQueryTerm = (input: string) => input.replace(/\\/g, '\\\\').replace(/\"/g, '\\"');
+const createQuery = (type: string, id: string, rawIdPrefix: string) =>
+  `"${createQueryTerm(`${rawIdPrefix}${type}:${id}`)}" | "${createQueryTerm(id)}"`;
+
+/**
+ * Query for a saved object with a given origin id and replace the
+ * id in the provided action with the _id from the query result
+ * @param action
+ * @param esClient
+ * @returns
+ */
+export const swapActionIds = async (
+  action: Action,
+  savedObjectsClient: SavedObjectsClientContract
+): Promise<Action | Error> => {
+  try {
+    // TODO: might need to pass down namespace
+    // but for now it's blank to do local testing.
+    const search = createQuery('action', action.id, '');
```

Comment:
The namespace prefix here is a bug that we didn't notice because a gap in our integration tests 🤦 I documented the bug in #122149 and I have a draft PR to fix it (and a few other import/copy bugs) in #121046. 
The specific commit with the fix is here: f69e2d41f96b20e3255585977ac9c98a6d00b294

You are already using a blank string for the namespace prefix so your code isn't affected 🙌 TL;DR we should never use a namespace prefix for multi-namespace object types, that was an accident.

You can make these changes:

```diff
diff --git a/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts b/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts
index e05a7f6394b..21afa378584 100644
--- a/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts
+++ b/x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts
@@ -198,8 +198,8 @@ export const getTupleDuplicateErrorsAndUniqueRules = (
 // functions copied from here
 // https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/server/saved_objects/import/lib/check_origin_conflicts.ts#L55-L57
 const createQueryTerm = (input: string) => input.replace(/\\/g, '\\\\').replace(/\"/g, '\\"');
-const createQuery = (type: string, id: string, rawIdPrefix: string) =>
-  `"${createQueryTerm(`${rawIdPrefix}${type}:${id}`)}" | "${createQueryTerm(id)}"`;
+const createQuery = (type: string, id: string) =>
+  `"${createQueryTerm(`${type}:${id}`)}" | "${createQueryTerm(id)}"`;
 
 /**
  * Query for a saved object with a given origin id and replace the
@@ -213,9 +213,7 @@ export const swapActionIds = async (
   savedObjectsClient: SavedObjectsClientContract
 ): Promise<Action | Error> => {
   try {
-    // TODO: might need to pass down namespace
-    // but for now it's blank to do local testing.
-    const search = createQuery('action', action.id, '');
+    const search = createQuery('action', action.id);
     const foundAction = await savedObjectsClient.find<Action>({
       type: 'action',
       search,
```